service/history/shard/controller_impl.go

```diff
@@ -304,7 +351,11 @@ func (c *ControllerImpl) acquireShards(ctx context.Context) {
 		if err := c.ownership.verifyOwnership(shardID); err != nil {
 			if IsShardOwnershipLostError(err) {
 				// current host is not owner of shard, unload it if it is already loaded.
-				c.CloseShardByID(shardID)
+				if c.config.ShardLingerEnabled() {
+					c.shardLingerThenClose(ctx, shardID)
```

Comment:
This is synchronous for this concurrency slot. So we can only have `AcquireShardConcurrency` (default 10) shards lingering at once? And while they are, we won't preemptively acquire any more? (Though we could acquire them based on incoming rpcs.) It seems like when there's movement, we both lose some and gain some, so we'd want that all to proceed in parallel. Do we need to turn up the concurrency to use this feature effectively?